dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -0,0 +1,80 @@
+package frameless.functions
+
+import frameless.{TypedColumn, TypedEncoder}
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.{udf => sudf}
+
+import scala.reflect.runtime.universe._
+
+trait Udf {
+  def udf[T, A: TypeTag, R: TypeTag](
+     f: A => R)
+  (implicit
+     aenc: TypedEncoder[A],
+     reng: TypedEncoder[R]):
+  (TypedColumn[T, A] => TypedColumn[T, R]) =
+    (u: TypedColumn[T, A]) => {
+      val uf = sudf(f)
```

Comment:
This is strange that we need `TypeTag`. As I can see from `udf` in Spark it uses `TypeTag` for
```scala
    val inputTypes = Try(ScalaReflection.schemaFor(typeTag[A1]).dataType :: Nil).toOption
```

It wouldn't use `TypedEncoder`, to make use of frameless encoders we have to use `UserDefinedFunction` directly, that takes a list of input types.